### PR TITLE
Apple TV support

### PR DESCRIPTION
**Pull request by douglowder**
**Motivation**

Adding support for Apple TV.  Please see the new file README-appletv.md for details.

**Test plan (required)**

Jest tests, Xcode unit tests, and integration tests pass.

### CODE REVIEW

**Comment by facebook-github-bot-6** [comment]
By analyzing the blame information on this pull request, we identified @adamjernst and @nathanajah to be potential reviewers.

**Comment by facebook-github-bot-6** [comment]
Thank you for your pull request and welcome to our community. We require contributors to sign our Contributor License Agreement, and we don't seem to have you on file. In order for us to review and merge your code, please sign up at https://code.facebook.com/cla - and if you have received this in error or have any questions, please drop us a line at cla@fb.com. Thanks!

**Comment by douglowder** [comment]
Sent a note to cla@fb.com about the Contributor License Agreement, hopefully that will be resolved shortly.

File: Examples/2048/Game2048.js [hunk 1]
```diff
@@ -29,9 +29,9 @@ var Animated = require('Animated');
 var GameBoard = require('GameBoard');
 var TouchableBounce = require('TouchableBounce');
 
-var BOARD_PADDING = 3;
-var CELL_MARGIN = 4;
-var CELL_SIZE = 60;
+var BOARD_PADDING = __APPLETV__ ? 6 : 3;
+var CELL_MARGIN = __APPLETV__ ? 8 : 4;
+var CELL_SIZE = __APPLETV__ ? 180 : 60;
```
Comment (douglowder) [inline Examples/2048/Game2048.js]: __APPLETV__ is defined when the packager builds the bundle.

**Comment by facebook-github-bot-8** [comment]
@douglowder updated the pull request - [view changes](https://github.com/facebook/react-native/pull/9533/files/90e143785fd5971d4e543a3fc9bf5fb6fbebf8d5..0426ac0cdf314ffc93014a5766ab00f9352eb795)

**Comment by SalahAdDin** [comment]
Great!

**Comment by facebook-github-bot-6** [comment]
@douglowder updated the pull request - [view changes](https://github.com/facebook/react-native/pull/9533/files/0426ac0cdf314ffc93014a5766ab00f9352eb795..696ba240cab7e2d857d47e90edeb10b52d8c5f04)

**Comment by facebook-github-bot-1** [comment]
@douglowder updated the pull request - [view changes](https://github.com/facebook/react-native/pull/9533/files/696ba240cab7e2d857d47e90edeb10b52d8c5f04..94fae68e9c77bf5de3f9d84910a34c5fde5efec4)

**Comment by facebook-github-bot-3** [comment]
@douglowder updated the pull request - [view changes](https://github.com/facebook/react-native/pull/9533/files/94fae68e9c77bf5de3f9d84910a34c5fde5efec4..61138a39b42a09c217b99fd95d7390df630aae47)

**Comment by satya164** [comment]
Instead of a global variable, how about adding something to `Platform` and using that?

**Comment by satya164** [comment]
BTW this is sooo amazing! Bravo!

**Comment by douglowder** [comment]
@satya164 I could have added a new Platform type, but it seemed to me that that would have required much more extensive changes, including a lot of test changes.  Since tvOS and iOS share so much code in common, it seemed easier to just add the global Apple TV flag.

**Comment by satya164** [comment]
`Platform.OS` could stay as `iOS`, and you could add something like `Platform.Variant` I think. Should need almost the same amount of changes as a global variable.

**Comment by douglowder** [comment]
So now it looks like Travis CI is passing, but Circle CI is failing.  The failure seems unrelated to my changes.  @adamjernst @nathanajah

**Comment by facebook-github-bot-7** [comment]
@douglowder updated the pull request - [view changes](https://github.com/facebook/react-native/pull/9533/files/61138a39b42a09c217b99fd95d7390df630aae47..d35a080f66109aef2b9babe7196efe24117688e2)

**Comment by douglowder** [comment]
Looks like this might have been fixed upstream, just merged from upstream and hopefully Circle CI passes now.

**Comment by douglowder** [comment]
Hmmm Travis seems to be stuck... I'll merge upstream and see if that kickstarts it....

**Comment by facebook-github-bot-9** [comment]
@douglowder updated the pull request - [view changes](https://github.com/facebook/react-native/pull/9533/files/d35a080f66109aef2b9babe7196efe24117688e2..abc2af0eead376a37400a3e56df8632f3e94ed70)

**Comment by facebook-github-bot-6** [comment]
@douglowder updated the pull request - [view changes](https://github.com/facebook/react-native/pull/9533/files/abc2af0eead376a37400a3e56df8632f3e94ed70..c174cc971002a3ef5047515bb003a613ceecfd65)

**Comment by facebook-github-bot-9** [comment]
@douglowder updated the pull request - [view changes](https://github.com/facebook/react-native/pull/9533/files/c174cc971002a3ef5047515bb003a613ceecfd65..8cbb8cf7caec7871d4830760ee1cef965618eaef)

**Comment by satya164** [comment]
cc @mkonicek @bestander

**Comment by douglowder** [comment]
Closing this pull request and opening a new one.

**Comment by dlowder-salesforce** [comment]
New pull request created: https://github.com/facebook/react-native/pull/9577

**Pull request closed without merging**